### PR TITLE
DCS-774 Fixing issue around nullability and end dates

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/CourtService.kt [hunk 1]
```diff
@@ -246,14 +252,20 @@ class CourtService(
...
+    scheduledAppointment?.takeIf { hasAnEndDate(it) }?.let {
       VideoLinkBookingResponse.LocationTimeslot(
         locationId = it.locationId,
         startTime = it.startTime,
-        endTime = it.endTime
+        endTime = it.endTime!!
       )
     }
 
+  private fun hasAnEndDate(it: ScheduledAppointmentDto): Boolean {
```
Comment (bell-pa) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/CourtService.kt:264]: Could be an extension function. Don't need it though.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/services/CourtServiceTest.kt [hunk 2]
```diff
@@ -1047,6 +1047,74 @@ class CourtServiceTest {
...
+            VideoLinkBooking(
+              id = 1000L,
+              pre = VideoLinkAppointment(
+                id = 10001,
+                bookingId = 1000,
+                appointmentId = 1001,
+                court = "Wimbledon",
+                hearingType = HearingType.PRE
+              ),
+              main = VideoLinkAppointment(
```
Comment (bell-pa) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/services/CourtServiceTest.kt:1094]: What happens if the main appointment isn't valid? 
Remove the booking from the result list?

**Review by bell-pa — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/services/CourtServiceTest.kt (hunk 2 above)
Comment (andrewrlee) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/services/CourtServiceTest.kt:1094]: fixed that

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/CourtService.kt (hunk 1 above)
Comment (andrewrlee) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/CourtService.kt:264]: ooh that would be nice

**Pull request merged**